Bottom: c62cb8957d75c6cd3208a365c197e2853e05e158
Top:    9b5bcaad5b09650901f3a3c17bd97f7d1fd204c7
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-30 18:34:51 +0800

autofs-5.0.7 - try and cleanup after dumpmaps

Try and cleanup a little after dumping maps.
It's not really necessary but but can help by reducing the noise
from valgrind when checking code.


---

diff --git a/CHANGELOG b/CHANGELOG
index cadf5d2..1cf79be 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -58,6 +58,7 @@
 - add after sssd dependency to unit file.
 - dont start readmap unless ready.
 - fix dumpmaps multi output.
+- try and cleanup after dumpmaps.
 
 25/07/2012 autofs-5.0.7
 =======================
diff --git a/daemon/automount.c b/daemon/automount.c
index 1d0b64e..6849272 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2136,6 +2136,8 @@ int main(int argc, char *argv[])
 	}
 
 	if (dumpmaps) {
+		struct master_mapent *entry;
+		struct list_head *head, *p;
 		struct mapent_cache *nc;
 
 		open_log();
@@ -2152,6 +2154,17 @@ int main(int argc, char *argv[])
 
 		lookup_nss_read_master(master_list, 0);
 		master_show_mounts(master_list);
+
+		head = &master_list->mounts;
+		p = head->next;
+		while (p != head) {
+			entry = list_entry(p, struct master_mapent, list);
+			p = p->next;
+			master_free_mapent_sources(entry, 1);
+			master_free_mapent(entry);
+		}
+		master_kill(master_list);
+
 		exit(0);
 	}
 
diff --git a/lib/master.c b/lib/master.c
index bbf4b4e..c560496 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -1427,6 +1427,8 @@ int master_show_mounts(struct master *master)
 			source = source->next;
 		}
 
+		lookup_close_lookup(ap);
+
 		printf("\n");
 	}
